### PR TITLE
feat/search

### PR DESCRIPTION
**Pull request by mogitrash**
#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 💡 Background and solution

image

#### 🎯 Related issues

[link](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/train-a/search.md)

### CODE REVIEW

**Review by NataRep — APPROVED** [review]
Thank you for the great job you have done!

File: src/app/features/trips/components/trips-filter/trips-filter.component.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  public searchResponse$ = this.store.select(selectLastSearchReponse);
+
+  public searchDate$ = this.store.select(selectSearchDate);
+
+  public searchDates!: Date[];
+
+  constructor(private store: Store) {}
+
+  ngOnInit(): void {
+    this.searchResponse$.subscribe((searchResponse) => {
```
Comment (Arthurchkeee) [inline src/app/features/trips/components/trips-filter/trips-filter.component.ts:25]: Please, don't forget about unsubscription

File: src/app/features/trips/components/trips-filter/trips-filter.component.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      this.searchDates = Object.keys(searchResponse)
+        .map((date) => {
+          return new Date(date);
+        })
+        .sort((a, b) => {
+          return a.getTime() - b.getTime();
+        });
+    });
+
+    this.searchDate$.subscribe((searchDate) => {
```
Comment (Arthurchkeee) [inline src/app/features/trips/components/trips-filter/trips-filter.component.ts:35]: and here too

File: src/app/pages/main-page/main-page.component.ts [hunk 3]
```diff
@@ -1,8 +1,23 @@
...
   styleUrl: './main-page.component.scss',
 })
-export class MainPageComponent {}
+export class MainPageComponent implements OnInit {
+  public searchResponses$ = this.store.select(selectSearchResponses);
+
+  public searchResponses!: SearchResponse[];
+
+  ngOnInit(): void {
+    this.searchResponses$.subscribe((responses) => {
```
Comment (Arthurchkeee) [inline src/app/pages/main-page/main-page.component.ts:17]: Ubscription without unsubscribe

File: src/app/features/trips/components/trips-search/trips-search.component.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+@Component({
+  selector: 'app-trips-search',
+  templateUrl: './trips-search.component.html',
+  styleUrl: './trips-search.component.scss',
+})
+export class TripsSearchComponent implements OnInit {
+  public onSearchResponse = output<Subject<SearchResponseDTO>>();
+
+  public stations$ = this.store.select(selectStations);
```
Comment (Arthurchkeee) [inline src/app/features/trips/components/trips-search/trips-search.component.ts:23]: Maybe public elements can be protected?

**Review by Arthurchkeee — CHANGES_REQUESTED** [review]
Good job! Pleace, check my comments

**Review by Aleksa013 — APPROVED** [review]
Great job!

**Pull request merged**